### PR TITLE
Feat(eos_cli_config_gen): Add AAA Authentication Policy Lockout

### LINKED ISSUE #1345: Setting aaa authentication policy lockout
### Enhancement summary

Additional coverage for `aaa_authentication`

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

I need to generate the following, which is not supported by the current data model, and I can only generate it using `eos_cli`:

```
aaa authentication policy lockout failure 3 window 300 duration 900
```

### Describe the solution you would like

Modify the corresponding data models accordingly:

```
aaa_authentication:
  login:
    default: < group group_name | local | none > < group group_name | local | none >
    console: < group group_name | local | none > < group group_name | local | none >
  enable:
    default: < group group_name | local | none > < group group_name | local | none >
  dot1x:
    default: < group group_name >
  policies:
    on_failure_log: < true | false >
    on_success_log: < true | false >
    local:
      allow_nopassword: < false | true >
    lockout:
      failure: <min-max>
      window: <min-max>
      duration: <min-max>
```

### Describe alternatives you have considered

using `eos_cli` to render the CLI directly

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ccsnw**
## Change Summary

Added AAA Auth Policy Lockout option

## Related Issue(s)

Fixes #1345

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
added policy lockout options

```
aaa_authentication:
  login:
    default: < group group_name | local | none > < group group_name | local | none >
    console: < group group_name | local | none > < group group_name | local | none >
  enable:
    default: < group group_name | local | none > < group group_name | local | none >
  dot1x:
    default: < group group_name >
  policies:
    on_failure_log: < true | false >
    on_success_log: < true | false >
    local:
      allow_nopassword: < false | true >
    lockout:
      failure: < 1-255 >
      duration: < 1-4294967295 >
      window: < 1-4294967295 >
```

## How to test
molecule run

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authentication.j2 [hunk 1]
```diff
@@ -22,4 +22,11 @@ aaa authentication policy on-success log
 {%     if aaa_authentication.policies.local.allow_nopassword is arista.avd.defined(true) %}
 aaa authentication policy local allow-nopassword-remote-login
 {%     endif %}
+{%     if aaa_authentication.policies.lockout.failure is arista.avd.defined and aaa_authentication.policies.lockout.duration is arista.avd.defined %}
+{%         set lockout_cli = "aaa authentication policy lockout failure " ~ aaa_authentication.policies.lockout.failure ~ " duration " ~ aaa_authentication.policies.lockout.duration %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authentication.j2]: this code is not super clear. I would first add the common part. Then do the test for the optional window command and then add the duration last. This way you don't have the same stuff multiple times.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Almost there. Please adjust the config template a bit.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by onurgashi — APPROVED** [review]
LGTM!

**Pull request merged**